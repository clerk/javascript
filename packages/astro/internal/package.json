{
  "main": "../dist/internal/index.js"
}
