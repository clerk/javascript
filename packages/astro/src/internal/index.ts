/**
 * The following code will be used in order to be injected as script via the astro integration.
 * F.e.
 *
 * injectScript('before-hydration', `...`)
 */

import { createClerkInstance } from './create-clerk-instance';
import { createInjectionScriptRunner } from './create-injection-script-runner';

const runInjectionScript = createInjectionScriptRunner(createClerkInstance);

export { runInjectionScript };

export { generateSafeId } from './utils/generateSafeId';
