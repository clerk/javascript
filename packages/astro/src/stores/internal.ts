import type {
  ActiveSessionResource,
  Clerk,
  ClientResource,
  InitialState,
  OrganizationResource,
  UserResource,
} from '@clerk/types';
import { atom, map } from 'nanostores';

export const $csrState = map<{
  isLoaded: boolean;
  client: ClientResource | undefined | null;
  user: UserResource | undefined | null;
  session: ActiveSessionResource | undefined | null;
  organization: OrganizationResource | undefined | null;
}>({
  isLoaded: false,
  client: null,
  user: null,
  session: null,
  organization: null,
});

export const $initialState = map<InitialState>();

// Use atom instead of `map` to prohibit key changes and allow only replacing the whole object
export const $clerk = atom<Clerk | null>(null);
