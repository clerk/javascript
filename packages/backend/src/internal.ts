export { constants } from './constants';
export { createRedirect } from './createRedirect';
export type { RedirectFun } from './createRedirect';

export type { CreateAuthenticateRequestOptions } from './tokens/factory';
export { createAuthenticateRequest } from './tokens/factory';

export { debugRequestState } from './tokens/request';

export type { AuthenticateRequestOptions, OrganizationSyncOptions } from './tokens/types';

export type { SignedInAuthObjectOptions, SignedInAuthObject, SignedOutAuthObject } from './tokens/authObjects';
export { makeAuthObjectSerializable, signedOutAuthObject, signedInAuthObject } from './tokens/authObjects';

export { AuthStatus } from './tokens/authStatus';
export type { RequestState, SignedInState, SignedOutState } from './tokens/authStatus';

export { decorateObjectWithResources, stripPrivateDataFromObject } from './util/decorateObjectWithResources';

export { createClerkRequest } from './tokens/clerkRequest';
export type { ClerkRequest } from './tokens/clerkRequest';

export { reverificationError, reverificationErrorResponse } from '@clerk/shared/authorization-errors';
