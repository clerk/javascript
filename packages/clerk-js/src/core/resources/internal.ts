export type { default as Clerk } from '../clerk';
export * from './Base';
export * from './UserSettings';
export * from './AuthConfig';
export * from './Client';
export * from './DeletedObject';
export * from './DisplayConfig';
export * from './EmailAddress';
export * from './Environment';
export * from './Error';
export * from './ExternalAccount';
export * from './IdentificationLink';
export * from './Image';
export * from './PhoneNumber';
export * from './Organization';
export * from './OrganizationInvitation';
export * from './OrganizationMembership';
export * from './SamlAccount';
export * from './Session';
export * from './SessionWithActivities';
export * from './SignIn';
export * from './SignUp';
export * from './Token';
export * from './TOTP';
export * from './User';
export * from './Verification';
export * from './Web3Wallet';
