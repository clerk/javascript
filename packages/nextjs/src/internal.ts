/**
 * These need to be explicitly listed. Do not use an * here.
 * If you do, app router will break.
 */
export { MultisessionAppSupport } from './client-boundary/controlComponents';
