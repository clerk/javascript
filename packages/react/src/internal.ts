import type { ErrorThrowerOptions } from '@clerk/shared/error';

import { errorThrower } from './errors/errorThrower';

/**
 * Overrides options of the internal errorThrower (eg setting packageName prefix).
 *
 * @internal
 */
export function setErrorThrowerOptions(options: ErrorThrowerOptions) {
  errorThrower.setMessages(options).setPackageName(options);
}
