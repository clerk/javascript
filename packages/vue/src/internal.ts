export { setErrorThrowerOptions } from './errors/errorThrower';
