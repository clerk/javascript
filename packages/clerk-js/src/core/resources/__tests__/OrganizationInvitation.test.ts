import { OrganizationInvitation } from '../internal';

describe('OrganizationInvitation', () => {
  it('has the same initial properties', () => {
    const organizationInvitation = new OrganizationInvitation({
      object: 'organization_invitation',
      email_address: 'test_email',
      id: 'test_id',
      organization_id: 'test_organization_id',
      public_metadata: {
        public: 'metadata',
      },
      role: 'basic_member',
      created_at: 12345,
      updated_at: 5678,
      status: 'pending',
    });

    expect(organizationInvitation).toMatchObject({
      id: 'test_id',
      emailAddress: 'test_email',
      organizationId: 'test_organization_id',
      role: 'basic_member',
      roleName: undefined,
      status: 'pending',
      createdAt: expect.any(Date),
      updatedAt: expect.any(Date),
      publicMetadata: {
        public: 'metadata',
      },
    });
  });
});
