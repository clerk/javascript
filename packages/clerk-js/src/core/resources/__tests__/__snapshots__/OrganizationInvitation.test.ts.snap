// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationInvitation has the same initial properties 1`] = `
OrganizationInvitation {
  "createdAt": 1970-01-01T00:00:12.345Z,
  "emailAddress": "test_email",
  "id": "test_id",
  "organizationId": "test_organization_id",
  "pathRoot": "",
  "publicMetadata": {
    "public": "metadata",
  },
  "revoke": [Function],
  "role": "basic_member",
  "status": "pending",
  "updatedAt": 1970-01-01T00:00:05.678Z,
}
`;
